### PR TITLE
8252000: remove usage of PropertyResolvingWrapper in vmTestbase/nsk/jdb

### PR DESCRIPTION
**Pull request by iignatev**
the patch
- removes `PropertyResolvingWrapper` from `vmTestbase/nsk/jdb` tests
- updates `JdbArgumentHandler` to remove `"` from `jdb.option` option
- reformats code

testing:
✅ `vmTestbase/nsk/jdb` on {macosx,linux,windows}-x64

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8252000](https://bugs.openjdk.java.net/browse/JDK-8252000): remove usage of PropertyResolvingWrapper in vmTestbase/nsk/jdb

### Reviewers
 * [Aleksey Shipilev](https://openjdk.java.net/census#shade) (@shipilev - **Reviewer**) ⚠️ Review applies to 2cd3ef188433225139e90fef3bf71cb9f3170451
 * [Chris Plummer](https://openjdk.java.net/census#cjplummer) (@plummercj - **Reviewer**)
 * [Serguei Spitsyn](https://openjdk.java.net/census#sspitsyn) (@sspitsyn - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/350/head:pull/350`
`$ git checkout pull/350`

### CODE REVIEW

File: test/hotspot/jtreg/vmTestbase/nsk/jdb/clear/clear004/clear004.java [hunk 1]
```diff
@@ -110,19 +109,18 @@ protected void runCases() {
...
-    private boolean checkBreakpoint (String breakpoint, Paragrep grep) {
+    private boolean checkBreakpoint(String breakpoint, Paragrep grep) {
         String found;
         boolean result = true;
-        int count;
-        Vector v;
+        Vector<String> v;
 
-        v = new Vector();
+        v = new Vector<>();
```
Comment (shipilev) [inline test/hotspot/jtreg/vmTestbase/nsk/jdb/clear/clear004/clear004.java:123]: Coalesce these two lines into `Vector<String> v = new Vector<>()`?

File: test/hotspot/jtreg/vmTestbase/nsk/jdb/clear/clear004/clear004.java [hunk 2]
```diff
@@ -134,15 +132,14 @@ private boolean checkBreakpoint (String breakpoint, Paragrep grep) {
...
         Paragrep grep;
         String found;
         String[] reply;
         boolean result = true;
-        int count;
-        Vector v;
+        Vector<String> v;
 
-        v = new Vector();
+        v = new Vector<>();
```
Comment (shipilev) [inline test/hotspot/jtreg/vmTestbase/nsk/jdb/clear/clear004/clear004.java:142]: `Vector<String> v = new Vector<>()`?

File: test/hotspot/jtreg/vmTestbase/nsk/jdb/locals/locals002/locals002.java [hunk 3]
```diff
@@ -88,72 +90,58 @@ public static int run(String argv[], PrintStream out) {
...
+            {"boolVar", "true", "false"},
+            {"byteVar", "27", "12"},
+            {"charVar", "V", "A"},
+            {"shortVar", "767", "327"},
+            {"intVar", "1474", "3647"},
+            {"longVar", "21345", "65789"},
+            {"floatVar", "3.141", "4.852"},
+            {"doubleVar", "2.578", "3.8976"},
+            {"objVar", "objVarString", "objArgString"},
+            {"arrVar", "int[5]", "int[3]"}
```
Comment (shipilev) [inline test/hotspot/jtreg/vmTestbase/nsk/jdb/locals/locals002/locals002.java:109]: This table looked a bit better before. Maybe commas should move to the left, but tabbing restored?

File: test/hotspot/jtreg/vmTestbase/nsk/jdb/regression/b4689395/b4689395.java [hunk 4]
```diff
@@ -114,58 +115,59 @@
...
+    }
 
-        public b4689395() {
-                classFile = ClassLoadUtils.getRedefineClassFileName(DEBUGGEE_CLASS);
-                if (classFile == null)
-                        throw new TestFailure("Unable to find redefine class file in classpath for: " + DEBUGGEE_CLASS);
-        }
+    public b4689395() {
+        classFile = ClassLoadUtils.getRedefineClassFileName(DEBUGGEE_CLASS);
+        if (classFile == null)
```
Comment (shipilev) [inline test/hotspot/jtreg/vmTestbase/nsk/jdb/regression/b4689395/b4689395.java:147]: Braces, while we are changing these lines anyway?

File: test/hotspot/jtreg/vmTestbase/nsk/jdb/caught_exception/caught_exception002/caught_exception002.java [hunk 5]
```diff
@@ -112,11 +108,10 @@ protected void runCases() {
         jdb.contToExit(1);
     }
 
-    private void checkCatch (String[] reply, int i) {
+    private void checkCatch(String[] reply, int i) {
         Paragrep grep;
         int count;
-        String found;
-        Vector v = new Vector();
+        Vector<String> v = new Vector<>();
```
Comment (shipilev) [inline test/hotspot/jtreg/vmTestbase/nsk/jdb/caught_exception/caught_exception002/caught_exception002.java:114]: As long as we cleaning up the coding here, the declarations can be moved to the first use? Here... and everywhere.

**Review by shipilev — APPROVED** [review]
The changes look good, but I think the practice of doing a massive cleanup in "remove usage of PropertyResolvingWrapper in vmTestbase/nsk/jdb" PR is misleading. Better to split this in two issues (at least next time)?

**Comment by iignatev** [comment]
> The changes look good, but I think the practice of doing a massive cleanup in "remove usage of PropertyResolvingWrapper in vmTestbase/nsk/jdb" PR is misleading. Better to split this in two issues (at least next time)?

o/c I can split, however, given I'm not (yet?) on the crusade to clean up all the code within `test/hotspot` and just do clean up in the touched files, the title for such an RFE would be somewhat weird and/or misleading, e.g. 'clean up code in the files Igor touched in 8252000', 'some cleans up in some files within vmTestbase/nsk/jdb', etc.

**Comment by iignatev** [comment]
@shipilev , the next iteration should have all your comments addressed.

**Comment by iignatev** [comment]
@shipilev , do you have any further comments? Or are you (relatively) happy w/ the patch after the last 3 commits?

**Comment by plummercj** [comment]
I have to push back on this. Needs to be broken up. You have 3 large changes that are completely unrelated. Seems you can push any of those 3 independent of the others.

**Comment by iignatev** [comment]
> I have to push back on this. Needs to be broken up. You have 3 large changes that are completely unrelated. Seems you can push any of those 3 independent of the others.

Hi Chris,

what do you see as a 3rd change here?

`removes PropertyResolvingWrapper from vmTestbase/nsk/jdb tests` and `updates JdbArgumentHandler to remove " from jdb.option option` parts aren't independent. the only independent part is code reformating, which, at least in my book, would include moving var declaration. so if you insist I can split into two patches: [5f62042](https://github.com/openjdk/jdk/pull/350/commits/5f62042817700b1240cbbd29a56523eb00b6b084) (+ copyright year updates if any) and the rest.

-- Igor

**Comment by plummercj** [comment]
> what do you see as a 3rd change here?

PropertyResolvingWrapper, format changes, var declaration changes. I think large format changes should be kept separate from other changes, and would preferably be done first.

**Comment by iignatev** [comment]
> > what do you see as a 3rd change here?
> 
> PropertyResolvingWrapper, format changes, var declaration changes. I think large format changes should be kept separate from other changes, and would preferably be done first.

ok, given we are already in PR to remove PropertyResolvingWrapper, I suggest go w/ it 1st and I'll move other changes into separate RFE/fixes/PRs.

**Review by plummercj — APPROVED** [review]
The changes look good, but can you clarify (preferably in the CR) why the getJdbOptions() changes are needed when removing the use of PropertyResolvingWrapper?

**Review by sspitsyn — APPROVED** [review]

**Comment by sspitsyn** [comment]
Igor, this update looks good to me.

**Comment by iignatev** [comment]
Chris, Serguei, thanks for your reviews. I've added the comment[[*]] explaining the accompanying changes, so should someone need to dig up why these changes were done, they will get some background. please let me know if it needs to be more elaborate.

/integrate

[*]: https://bugs.openjdk.java.net/browse/JDK-8252000?focusedCommentId=14371301&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-14371301

**Pull request closed without merging**